python/fledge/services/core/api/plugins/remove.py

```diff
@@ -0,0 +1,215 @@
+                        plugin_users.append({'e': e, 'service_list': service_list})
+                    else:
+                        service_list.append(e['service'])
+                        plugin_users.append({'e': e, 'service_list': service_list})
+    return plugin_users
+
+
+async def check_service_in_schedules(service_name: str):
+    storage_client = connect.get_storage_async()
+    payload_data = PayloadBuilder().SELECT('id', 'enabled').WHERE(['schedule_name', '=', service_name]).payload()
+    enabled_service_list = await storage_client.query_tbl_with_payload('schedules', payload_data)
+    is_service_list = True if enabled_service_list['rows'] else False
+    return is_service_list
+
+
+async def check_plugin_usage_in_notification_instances(plugin_name: str):
+    """ Check notification instance state using the given rule or delivery plugin
+    """
+    notification_instances = []
+    storage_client = connect.get_storage_async()
+    configuration_mgr = ConfigurationManager(storage_client)
+    notifications = await configuration_mgr.get_category_child("Notifications")
+    if notifications:
+        for notification in notifications:
+            notification_config = await configuration_mgr._read_category_val(notification['key'])
+            name = notification_config['name']['value']
+            channel = notification_config['channel']['value']
+            rule = notification_config['rule']['value']
+            enabled = True if notification_config['enable']['value'] == 'true' else False
+            if (channel == plugin_name and enabled) or (rule == plugin_name and enabled):
+                notification_instances.append(name)
+    return notification_instances
+
+
+def purge_plugin(plugin_type: str, name: str) -> tuple:
+    _logger.info("{} plugin removal started...".format(name))
+    original_name = name
+    # Special case handling - installed directory name Vs package name
+    # For example: Plugins like http_south Vs http-south
+    name = name.replace('_', '-').lower()
+    plugin_name = 'fledge-{}-{}'.format(plugin_type, name)
+    stdout_file_path = common.create_log_file(action='remove', plugin_name=plugin_name)
+    link = "log/" + stdout_file_path.split("/")[-1]
+    get_platform = platform.platform()
+    try:
+        if 'centos' in get_platform or 'redhat' in get_platform:
+            # TODO: Add yum based package / repo check
+            cmd = "sudo yum -y remove {} > {} 2>&1".format(plugin_name, stdout_file_path)
+        else:
+            dpkg_list = os.popen('dpkg --list fledge* 2>/dev/null')
+            ls_output = dpkg_list.read()
+            _logger.debug("dpkg list output: {}".format(ls_output))
+            if len(ls_output) :
+                f = ls_output.find(plugin_name)
+                if f == -1:
+                    raise KeyError
+            else:
+                raise KeyError
+            cmd = "sudo apt -y purge {} > {} 2>&1".format(plugin_name, stdout_file_path)
```

Comment:
Also we need to add this command in sudoers file. Like `/usr/bin/apt -y purge fledge*.deb`
https://github.com/fledge-iot/fledge/blob/15548a1b6a24f3d73ece8c3d136d403a7db6a809/scripts/extras/fledge.sudoers#L1